### PR TITLE
Smooth form with upgrading

### PR DESCRIPTION
**Pull request by kaspth**
- Default `config.action_view.embed_authenticity_token_in_remote_forms` to nil (so we can start embedding tokens in `form_with` by default), but have a past assignment of either false or true mean the same thing.
- Add `config.action_view.form_with_generates_remote_forms` to keep the past remote default
- Default 👆 to false in new_framework_defaults_5_1.rb.

### CODE REVIEW

**Comment by kaspth** [comment]
Gonna fix the tests tomorrow. Brain is part way cooked and I've half forgotten what this what meant to do in the first place 😅

File: actionview/lib/action_view/helpers/form_tag_helper.rb [hunk 1]
```diff
@@ -828,9 +828,8 @@ def html_options_for_form(url_for_options, options)
 
             html_options["data-remote"] = true if html_options.delete("remote")
 
-            if html_options["data-remote"] &&
-               !embed_authenticity_token_in_remote_forms &&
-               html_options["authenticity_token"].blank?
+            if html_options["data-remote"] && embed_authenticity_token_in_remote_forms == false &&
```
Comment (rafaelfranca) [inline actionview/lib/action_view/helpers/form_tag_helper.rb]: can we keep the style of the line?

File: railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_5_1.rb.tt [hunk 2]
```diff
@@ -5,6 +5,9 @@
 # Once upgraded flip defaults one by one to migrate to the new default.
 #
 # Read the Guide for Upgrading Ruby on Rails for more info on each option.
+
+# Make `form_with` generate non-remote forms.
+Rails.application.config.action_view.keep_form_with_remote = false
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_5_1.rb.tt]: What keep form with remote means?

That all forms generated with form with is generated with remote true?

Comment (kaspth) [inline railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_5_1.rb.tt]: > That all forms generated with form with is generated with remote true?

Yep, they are. Unless that config is turned off.

Comment (rafaelfranca) [inline railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_5_1.rb.tt]: So maybe change the config to `remote_forms = true`?

**Comment by y-yagi** [comment]
Should we update this check?
https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/app_generator.rb#L406..L410

**Comment by kaspth** [comment]
Going to move forward with this (fixed a bug around `form_with` and explicit `authenticity_token` values too) — thanks for the reviews!

**Comment by rafaelfranca** [comment]
Backported in d2e18641eeac1586f614a397880359fab5c3fdbf

**Pull request merged**